### PR TITLE
feat(RHINENG-5449): add new prometheus metrics

### PR DESCRIPTION
**Pull request by FabriciaDinizRH**
# Overview

This PR is being created to address [RHINENG-5449](https://issues.redhat.com/browse/RHINENG-5449).
This PR adds the following metrics to HBI: `consumed_message_size`, `produced_message_size` and `produce_large_message_failure`.

Since I'll be away, please feel free to merge commits with fixes if needed, so this PR can be merged as soon as possible.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/queue/metrics.py [hunk 1]
```diff
@@ -32,6 +32,11 @@
 ingress_message_handler_time = Summary(
     "inventory_ingress_message_handler_seconds", "Total time spent handling messages from the ingress queue"
 )
+consumed_message_size = Summary("inventory_consumed_message_size", "Size of incoming messages in bytes")
+produced_message_size = Summary("invenotry_produced_message_size", "Size of outgoing messages in bytes")
+produce_large_message_failure = Counter(
```
Comment (vkrizan) [inline app/queue/metrics.py:37]: It might be good to have counter of all produced messages to calculate a ratio. WDYT? If this is a temporary metric, then consider my comment irrelevant.

File: app/queue/event_producer.py [hunk 2]
```diff
@@ -22,9 +24,15 @@ def __init__(self, topic: str, event: str, headers: list(tuple()), key: str):
         self.topic = topic
 
     def on_delivered(self, error, message):
+        message_to_send = None
         if error:
-            message_not_produced(logger, error, self.topic, self.event, self.key, self.headers)
+            if "value too long" in error:
```
Comment (vkrizan) [inline app/queue/event_producer.py:29]: Same comment as in https://github.com/RedHatInsights/insights-host-inventory/pull/1598#discussion_r1478005242

File: app/queue/event_producer.py [hunk 3]
```diff
@@ -22,9 +24,15 @@ def __init__(self, topic: str, event: str, headers: list(tuple()), key: str):
...
+                message_to_send = message
```
Comment (vkrizan) [inline app/queue/event_producer.py:30]: This looks unrelated to the metrics. Looks like the #1598 work.

Comment (FabriciaDinizRH) [reply app/queue/event_producer.py:30]: yeah, I had to merge this before the other PR so that I could trigger the metric in the right place

**Review by kruai — APPROVED** [review]
